### PR TITLE
avm2: Partially implement Loader.load

### PR DESCRIPTION
**Pull request by Aaron1011**
Depends on https://github.com/ruffle-rs/ruffle/pull/7633

This PR implements the `Loader.load` method, as well as
the associated `LoaderInfo` properties and events.

We can now load in an external AVM2 SWf: it will be added
as a child of `Loader` object, and will render properly
to the screen.

Limitations:
* The only supported `URLRequest` property is `url`
* `LoaderContext` is not supported at all - we always use the default
  behavior
* Only `Loader.load` is implemented - we do not yet support unloading.
* We fire a plain 'Event' for the 'progress' event, instead of using
  the (not yet implemented) 'ProgressEvent' class

The main changes in this PR are:
* The AVM2 `Loader` class now has an associated display object,
  `LoaderDisplay`. This is basically a stub, and just renders
  its single child (if it exists).
* `LoaderStream::Stage` is renamed to `LoaderStream::NotYetLoaded`.
  This is used for both the `Stage` and an 'uninitialized'
  `Loader.contentLoaderInfo`. In both cases, certain properties throw
  errors, while others return actual values.
* The rust `Loader` manager now handles both AVM1 and AVM2 movie loads.

### CODE REVIEW

**Comment by adrian17** [comment]
> instead of using the (not yet implemented) 'ProgressEvent' class

It's implemented :)

**Comment by Aaron1011** [comment]
@adrian17 I've updated the implementation to correctly handle the "progress" event.

File: core/src/avm2/globals.rs
Comment (Herschel) [inline core/src/avm2/globals.rs:101]: Should this be `eventdispatcher` to match the naming convention?

(Separate from this PR, should we use `snake_case` the naming convention for all of these eventually?)

File: swf/src/read.rs
Comment (Herschel) [inline swf/src/read.rs:2592]: Seems like this isn't very helpful to print the signature? It would be gibberish for any non-SWF file.

**Review by Herschel — APPROVED** [review]

**Comment by Herschel** [comment]
Thank you!

File: core/src/avm2/globals.rs
Comment (Aaron1011) [reply core/src/avm2/globals.rs:101]: I think snake case would be better in the long run, but I'm fine with changing this to `eventdispatcher` for now.

File: core/src/avm2/globals/flash/display/loader.rs
Comment (adrian17) [inline core/src/avm2/globals/flash/display/loader.rs:63]: let's use `&Multiname::public("url")` in new code, this is the more direct construction

File: core/src/avm2/object/loaderinfo_object.rs
Comment (adrian17) [inline core/src/avm2/object/loaderinfo_object.rs:177]: side note: maybe we should push this into `dispatch_event`? As it's currently duplicated many times already, I think

File: core/src/display_object/movie_clip.rs
Comment (adrian17) [inline core/src/display_object/movie_clip.rs:969]: is this used anywhere?

File: core/src/avm2/globals/flash/display/loader.rs
Comment (Aaron1011) [reply core/src/avm2/globals/flash/display/loader.rs]: I've adjusted this PR to use `&Multiname::public` where applicable.

**Review by adrian17 — APPROVED** [review]
LGTM, but would still prefer if Kmeisthax or Mike took a look before merging.

File: core/src/avm2/object/loaderinfo_object.rs
Comment (Aaron1011) [reply core/src/avm2/object/loaderinfo_object.rs:177]: Yeah, I was thinking about doing that in a follow-up PR.

File: core/src/display_object/movie_clip.rs
Comment (Aaron1011) [reply core/src/display_object/movie_clip.rs]: It's not - I've removed it.

**Pull request merged**
